torch/csrc/cuda/Event.cpp

```diff
@@ -37,27 +31,50 @@ static PyObject * THCPEvent_pynew(
   }
 
   THCPEvent* self = (THCPEvent *)ptr.get();
-  if (handle_bytes) {
-    AT_CHECK(handle_size == sizeof(cudaIpcEventHandle_t),
-      "cudaIpcEventHandle_t expects byte-like object of size ",
-      sizeof(cudaIpcEventHandle_t), ", but got ", handle_size);
-    AT_CHECK(device_index >= 0, "Reconstructing event from handle requires "
-      "a non-negtive device index, but got ", device_index)
-    // no need to release the handle byte array as it is automatically managed
-    // by the corresponding THCPEvent python object.
-    // see https://docs.python.org/3/c-api/arg.html#strings-and-buffers
-    cudaIpcEventHandle_t handle;
-    std::memcpy(&handle, handle_bytes, sizeof(handle));
-    new (&self->cuda_event) at::cuda::CUDAEvent(device_index, &handle);
-  } else {
-    unsigned int flags =
-      (blocking ? cudaEventBlockingSync : cudaEventDefault) |
-      (enable_timing ? cudaEventDefault : cudaEventDisableTiming) |
-      (interprocess ? cudaEventInterprocess : cudaEventDefault);
-
-    new (&self->cuda_event) at::cuda::CUDAEvent(flags);
+  unsigned int flags =
+    (blocking ? cudaEventBlockingSync : cudaEventDefault) |
+    (enable_timing ? cudaEventDefault : cudaEventDisableTiming) |
+    (interprocess ? cudaEventInterprocess : cudaEventDefault);
+
+  new (&self->cuda_event) at::cuda::CUDAEvent(flags);
+
+  return (PyObject *)ptr.release();
+  END_HANDLE_TH_ERRORS
+}
+
+static PyObject * THCPEvent_from_ipc_handle(
+    PyTypeObject *type, PyObject *args) {
+  HANDLE_TH_ERRORS
+  int64_t device_index = -1;
```

Comment:
This should be declared `long long` to match `L` in PyArg_ParseTuple. You can cast it later to `int64_t` at the use-site (`at::cuda::CUDAEvent(device_index, &handle);`)